collectors/mmlsqos.go

```diff
@@ -0,0 +1,228 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package collectors
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	qosFilesystems = kingpin.Flag("collector.mmlsqos.filesystems", "Filesystems to query with mmlsqos, comma separated. Defaults to all filesystems.").Default("").String()
+	qosTimeout     = kingpin.Flag("collector.mmlsqos.timeout", "Timeout for mmlsqos execution").Default("60").Int()
+	qosSeconds     = kingpin.Flag("collector.mmlsqos.seconds", "Display the I/O performance values for the previous number of seconds. The valid range of seconds is 1-999").Default("60").Int()
+	qosMap         = map[string]string{
+		"pool":      "Pool",
+		"timeEpoch": "Time",
+		"class":     "Class",
+		"iops":      "Iops",
+		"ioql":      "AvegarePendingRequests",
+		"qsdl":      "AvegareQueuedRequests",
+		"et":        "MeasurementInterval",
+		"MBs":       "Bs",
+	}
+	MmlsqosExec = mmlsqos
+)
+
+type QosMetric struct {
+	Pool                   string
+	Time                   float64
+	Class                  string
+	Iops                   float64
+	AvegarePendingRequests float64
+	AvegareQueuedRequests  float64
+	MeasurementInterval    float64
+	Bs                     float64
+}
+
+type MmlsqosCollector struct {
+	Iops                   *prometheus.Desc
+	AvegarePendingRequests *prometheus.Desc
+	AvegareQueuedRequests  *prometheus.Desc
+	MeasurementInterval    *prometheus.Desc
+	Bs                     *prometheus.Desc
+	logger                 log.Logger
+}
+
+func init() {
+	registerCollector("mmlsqos", false, NewMmlsqosCollector)
+}
+
+func NewMmlsqosCollector(logger log.Logger) Collector {
+	labels := []string{"fs", "pool", "class", "measurement_period_seconds"}
+	return &MmlsqosCollector{
+		Iops: prometheus.NewDesc(prometheus.BuildFQName(namespace, "qos", "iops"),
+			"GPFS performance of the class in I/O operations per second", labels, nil),
+		AvegarePendingRequests: prometheus.NewDesc(prometheus.BuildFQName(namespace, "qos", "average_pending_requests"),
+			"GPFS average number of I/O requests in the class that are pending for reasons other than being queued by QoS", labels, nil),
+		AvegareQueuedRequests: prometheus.NewDesc(prometheus.BuildFQName(namespace, "qos", "average_queued_requests"),
+			"GPFS average number of I/O requests in the class that are queued by QoS", labels, nil),
+		MeasurementInterval: prometheus.NewDesc(prometheus.BuildFQName(namespace, "qos", "measurement_interval_seconds"),
+			"GPFS interval in seconds during which the measurement was made", labels, nil),
+		Bs: prometheus.NewDesc(prometheus.BuildFQName(namespace, "qos", "bs"),
```

Comment:
```suggestion
		Bs: prometheus.NewDesc(prometheus.BuildFQName(namespace, "qos", "bytes_per_second"),
```

Reply:
Thanks! I've changed `bs` to `bytes_per_second`